rtuclient.go

```diff
@@ -276,27 +278,38 @@ func (mb *rtuSerialTransporter) Send(aduRequest []byte) (aduResponse []byte, err
 	// function := aduRequest[1]
 	// functionFail := aduRequest[1] & 0x80
 	bytesToRead := calculateResponseLength(aduRequest)
-	time.Sleep(mb.calculateDelay(len(aduRequest) + bytesToRead))
+	time.Sleep(mb.frameDelay() + mb.charsDuration(float64(len(aduRequest)+bytesToRead)))
 
 	data, err := readIncrementally(aduRequest[0], aduRequest[1], mb.port, time.Now().Add(mb.Config.Timeout))
 	mb.logf("modbus: recv % x\n", data[:])
 	aduResponse = data
 	return
 }
 
-// calculateDelay roughly calculates time needed for the next frame.
-// See MODBUS over Serial Line - Specification and Implementation Guide (page 13).
-func (mb *rtuSerialTransporter) calculateDelay(chars int) time.Duration {
-	var characterDelay, frameDelay int // us
+// charBits returns the number of bits per character.
+func (mb *rtuSerialTransporter) charBits() int {
+	var parityBits int
+	if mb.Parity != "N" {
+		parityBits++
+	}
+	return 1 + mb.DataBits + parityBits + mb.StopBits
+}
 
+// charDuration returns time needed for a character to be transmitted.
+func (mb *rtuSerialTransporter) charsDuration(chars float64) time.Duration {
+	return time.Duration(chars*float64(mb.charBits())/float64(mb.BaudRate)) * time.Second
```

Comment:
```suggestion
	return time.Duration(chars*mb.charBits) * time.Second / mb.BaudRate
```

Geht ohne `float64`.

Reply:
Gibt ein type missmatch. `chars` ist _float64_.